lib/functions.php

```diff
@@ -193,6 +193,10 @@ function ($matches) {
         $uri
     );
 
+    if (null === $uri) {
```

Comment:
Note: I don't see how to make a test case for this. I have to pass a string in `$uri`, and I can't find a string that will make this happen. Passing in "strings" like `\0` (a binary 0 byte) or `\n` (line terminator) etc does not cause `preg_replace_callback` to return `null`.

I suspect that this would be unreachable code - if the caller passes a non-string (int, bool...) in `$uri` then PHP will stop at the function call and report the invalid type passed. And any binary-string can be passed to `preg_replace_callback`

This is the challenge of code-analyzers - they need to be very aware of the possible state of variables and how all the functions exactly behave.